src/cpp/session/modules/SessionAsyncPackageInformation.cpp

```diff
@@ -53,23 +56,89 @@ class CompleteUpdateOnExit : public boost::noncopyable {
       using namespace rstudio::core::r_util;
 
       // Give empty completions to the packages which weren't updated
-      for (std::vector<std::string>::const_iterator it = AsyncRCompletions::s_pkgsToUpdate_.begin();
-           it != AsyncRCompletions::s_pkgsToUpdate_.end();
+      for (std::vector<std::string>::const_iterator it = AsyncPackageInformationProcess::s_pkgsToUpdate_.begin();
+           it != AsyncPackageInformationProcess::s_pkgsToUpdate_.end();
            ++it)
       {
-         if (!RSourceIndex::hasCompletions(*it))
+         if (!RSourceIndex::hasInformation(*it))
          {
-            RSourceIndex::addCompletions(*it, AsyncLibraryCompletions());
+            RSourceIndex::addPackageInformation(*it, PackageInformation());
          }
       }
 
-      AsyncRCompletions::s_pkgsToUpdate_.clear();
-      AsyncRCompletions::s_isUpdating_ = false;
+      AsyncPackageInformationProcess::s_pkgsToUpdate_.clear();
+      AsyncPackageInformationProcess::s_isUpdating_ = false;
    }
 
 };
 
-void AsyncRCompletions::onCompleted(int exitStatus)
+namespace {
+
+void fillFormalInfo(const json::Object& formalInfoJson,
+                    FunctionInformation* pInfo)
+{
+   using namespace core::json;
+   for (json::Object::const_iterator it = formalInfoJson.begin();
+        it != formalInfoJson.end();
+        ++it)
+   {
+      std::string formalName = it->first;
+      FormalInformation info(formalName);
+      
+      JSON_CHECK_TYPE(it->second, json::ObjectType);
+      const json::Object& fieldsJson = it->second.get_obj();
+      
+      info.setIsUsed(             JSON_INT_FIELD(fieldsJson, "is_used", 0));
```

Comment:
I think in this case, because I'm now reading from a JSON object with just plain `int` types that this will be appropriate (I hadn't realized that at the time). Let me clean up the code I added and refactor to just use `readObject`.